### PR TITLE
Update the Web API after the data/table/tasks refactoring

### PR DESCRIPTION
**Pull request by garaud**
depend on #24 

* fix several controller functions
* add unit test to test the API

### CODE REVIEW

File: jitenshea/controller.py [hunk 1]
```diff
@@ -180,100 +178,65 @@ def stations(city, limit, geojson):
...
-    if city == 'lyon':
-        return 'pvostationvelov', 'idstation'
+    return """SELECT id
+      ,name
+      ,address
+      ,city
+      ,nb_stations as nb_bikes
+      ,st_x(geom) as x
+      ,st_y(geom) as y
+    FROM {schema}.stations
```
Comment (delhomer) [inline jitenshea/controller.py]: We could recover the `stations` table name from config file, as in the previous PR.

File: jitenshea/controller.py [hunk 2]
```diff
@@ -180,100 +178,65 @@ def stations(city, limit, geojson):
...
         raise ValueError("City '{}' not supported.".format(city))
-    table, idcol = station_city_table(city)
     return """SELECT id
            ,number AS value
            ,date
-           ,Y.nom AS name
+           ,name
         FROM {schema}.daily_transaction AS X
-        LEFT JOIN {schema}.{table} AS Y ON X.id=Y.{idcol}::int
+        LEFT JOIN {schema}.stations AS Y using(id)
```
Comment (delhomer) [inline jitenshea/controller.py]: Same remark as previously.

File: jitenshea/controller.py [hunk 3]
```diff
@@ -296,14 +258,12 @@ def daily_query_stations(city, limit, order_by='station'):
         SELECT S.id
           ,D.number AS value
           ,D.date
-          ,Y.nom AS name
+          ,Y.name
         FROM station AS S
         LEFT JOIN {schema}.daily_transaction AS D ON (S.id=D.id)
-        LEFT JOIN {schema}.{table} AS Y ON S.id=Y.{idcol}::int
+        LEFT JOIN {schema}.stations AS Y ON S.id=Y.id
```
Comment (delhomer) [inline jitenshea/controller.py]: Idem

File: jitenshea/controller.py [hunk 4]
```diff
@@ -353,18 +313,23 @@ def daily_transaction_list(city, day, limit, order_by, window=0, backward=True):
...
 def timeseries(city, station_ids, start, stop):
     """Get timeseries data between two dates for a specific city and a list of station ids
     """
-    query = """SELECT *
-    FROM {schema}.timeserie_norm
-    WHERE id IN %(id_list)s AND ts >= %(start)s AND ts < %(stop)s
+    query = """SELECT T.*
+      ,S.name as name
+    FROM {schema}.timeseries AS T
+    LEFT JOIN {schema}.stations AS S using(id)
```
Comment (delhomer) [inline jitenshea/controller.py]: Two new occurrences of table names there (`timeseries` and `stations`).

File: jitenshea/controller.py [hunk 5]
```diff
@@ -466,9 +433,8 @@ def get_station_ids(city):
     list of integers
         IDs of the shared-bike stations in the `city`
     """
-    table, idcol = station_city_table(city)
-    query = ("SELECT {id} FROM {schema}.{table}"
-             ";").format(id=idcol, schema=config[city]["schema"], table=table)
+    query = ("SELECT id FROM {schema}.stations"
```
Comment (delhomer) [inline jitenshea/controller.py]: `stations` table could be automatized as well.

File: jitenshea/controller.py [hunk 6]
```diff
@@ -491,28 +457,25 @@ def station_cluster_query(city):
...
             "cs.start AS start, "
             "cs.stop AS stop, "
-            "citystation.nom AS nom, "
+            "citystation.name AS name, "
             "citystation.geom AS geom, "
             "rank() OVER (ORDER BY stop DESC) AS rank "
             "FROM {schema}.{cluster} AS cs "
-            "JOIN {schema}.{table} AS citystation "
-            "ON citystation.{idcol} = cs.station_id::varchar "
+            "JOIN {schema}.stations AS citystation "
```
Comment (delhomer) [inline jitenshea/controller.py]: Here comes a new occurrence.

**Comment by garaud** [comment]
thanks for the review. I updated the controller.py with your remarks.

**Pull request merged**